src/backend/executor/nodeModifyTable.c

```diff
@@ -632,7 +632,17 @@ ExecInsert(TupleTableSlot *parentslot,
 	}
 	if (canSetTag)
 	{
-		(estate->es_processed)++;
+		/*
+		 * If table is replicated, update es_processed only at one segment.
+		 * It allows not to adjust es_processed at QD after all executors send
+		 * the same value of es_processed.
+		 */
+		struct GpPolicy * cdbpolicy = resultRelInfo->ri_RelationDesc->rd_cdbpolicy;
+		if (Gp_role != GP_ROLE_EXECUTE || !GpPolicyIsReplicated(cdbpolicy) ||
+		    GpIdentity.segindex == (gp_session_id % cdbpolicy->numsegments))
```

Comment:
avoid local var for policy, according to https://github.com/arenadata/gpdb/commit/0bfc72517d7a28d5a4143ba026f57e5f35caf29e